app/src/main/java/org/mozilla/focus/shortcut/IconGenerator.kt

```diff
@@ -65,28 +59,37 @@ class IconGenerator {
         /**
          * Generates a launcher icon for versions of Android that support Adaptive Icons (Oreo+):
          * https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive.html
-         *
-         * A bitmap created from this drawable (e.g. Drawable.toBitmap) is expected to be used with
-         * [android.support.v4.graphics.drawable.IconCompat.createWithBitmap], not
-         * [android.support.v4.graphics.drawable.IconCompat.createWithAdaptiveBitmap]. Why? createWithAdaptiveBitmap
-         * will apply the platform's adaptive icon formatting to the given bitmap so it assumes you have a bitmap that
-         * hasn't been formatted already. However, when a bitmap is created with [AdaptiveIconDrawable], the formatting
-         * is applied so it'd be incorrect for us to use createWithAdaptiveBitmap, which would apply the formatting
-         * again.
-         *
-         * Unfortunately, we also can't just pass our legacy icon into createWithAdaptiveBitmap because the icon
-         * background has the corners removed and some device might display icons that have less of the corner missing
-         * than we do. In that case, we'd be unable to add the corners back and the icon wouldn't conform.
          */
         @TargetApi(26)
-        private fun generateAdaptiveLauncherIcon(context: Context, url: String?): AdaptiveIconDrawable {
+        private fun generateAdaptiveLauncherIcon(context: Context, url: String?): Bitmap {
```

Comment:
Most of this code is duplicated from `generateIcon`: can we merge the implementations? Duplication is bad because if we update one implementation, we often forget to update the others!